lib/endpoints/class-wp-rest-users-controller.php

```diff
@@ -841,6 +858,51 @@ public function get_collection_params() {
 			'type'               => 'string',
 			'validate_callback'  => 'rest_validate_request_arg',
 		);
+		$query_params['roles']   = array(
+			'description'        => __( 'Limit result set to resources matching at least one specific role provided. Accepts csv list or single role.' ),
+			'type'               => 'array',
+			'sanitize_callback'  => array( $this, 'rest_sanitize_user_roles' ),
+			'validate_callback'  => array( $this, 'rest_validate_user_roles' ),
+		);
 		return $query_params;
 	}
+
+	/**
+	 * Return a sanitized array of user roles.
+	 *
+	 * @param  mixed $roles Either an array of roles or an array
+	 * @return array $roles Sanitized user roles.
+	 */
+	public function rest_sanitize_user_roles( $roles ) {
+		if ( ! empty( $roles ) ) {
+			$roles = explode( ',', $roles );
+			foreach ( $roles as $role ) {
+				$role = sanitize_text_field( $role );
+				$sanitized_roles[] = $role;
+			}
+			return $sanitized_roles;
+		}
+		return $roles;
+	}
+
+	/**
+	 * Get the query params for collections
+	 *
+	 * @global $wp_roles Used by get_editable_roles.
+	 *
+	 * @param  array $roles User roles.
+	 * @return boolean|WP_Error
+	 */
+	public function rest_validate_user_roles( $roles ) {
```

Comment:
Permissions happens after as it does with every other request. I was just trying to keep things consistent.  Is that something that should be addressed in the infrastructure?  It would make more sense to me to check permissions for a request before anything is done, but I am also not an expert on that sort of stuff and if the current flow is better should I just chop the sanitization and validation out?.